Sources/MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -24,10 +24,18 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
     public lazy var directions: Directions = routingProvider as? Directions ?? Directions.shared
     
     /**
-     Routing provider used to create the route.
+     Custom `RoutingProvider`, used to create route.
+     
+     If set to `nil` - default implementation will be used.
      */
-    public var routingProvider: RoutingProvider
+    public var routingProvider: RoutingProvider?
```

Comment:
This is a breaking change. Can we mitigate it by setting it to `defaultRoutingProvider` by default? `defaultRoutingProvider` wouldn’t be lazy anymore, but can we make `MapboxRoutingProvider`’s members lazy so that creating it with `NavigationSettings.shared.navigationRouterType` is really cheap?

Reply:
Optional `RoutingProvider` is the entry point for rerouting customization. We cannot have 'default' provider set because technically we don't have such (it's buried inside NN). Of course LegacyRouteController doesn't rely on NN, so there is a default one, but since it's a `Router` implementation same as `RouteController`, we are forced to have optional value here too.

P.S.
Yes, there is `defaultRoutingProvider` in `RouteController` too, but it exists only because we still didn't integrate NN's refreshing mechanism. This will go away soon, so relying on it will just delay the inevitable.

Reply:
I agree, this PR introduces breaking changes that we have to avoid.
What if we create `internal final class DefaultRoutingProvider` that hides this lazy initialization of `MapboxRoutingProvider` and use it instead of `nil`?

Reply:
Another point here is that if we use a `default` here, it will create a false assumption that this entity is actually used for the reroute. But in fact, if user did not provide a custom `RoutingProvder` we won't use any at all, relying on NN reroute mechanism instead.
Thus, default `MapboxRoutingProvider` and customly set `MapboxRoutingProvider` will result in different behavior.

Reply:
The DefaultRoutingProvider suggestion would address the potential confusion, because it would be a different class that the developer can’t set themselves. It would be a mere façade for what the navigator does under the hood.